internal/groupsuffix/groupsuffix.go

```diff
@@ -101,6 +106,50 @@ func mutateOwnerRefs(replaceFunc func(baseAPIGroup, apiGroupSuffix string) (stri
 	}
 }
 
+func mutateTokenCredentialRequestAuthenticatorAPIGroup(apiGroupSuffix string) func(context.Context, kubeclient.RoundTrip) {
+	return func(_ context.Context, rt kubeclient.RoundTrip) {
+		// we only care if this is a TokenCredentialRequest
+		if rt.Resource() != loginv1alpha1.SchemeGroupVersion.WithResource("tokencredentialrequests") {
+			return
+		}
+
+		// we only do this on the way out, since on the way back in we don't set a spec in our
+		// TokenCredentialRequest
+		rt.MutateRequest(func(obj kubeclient.Object) {
+			tokenCredentialRequest, ok := obj.(*loginv1alpha1.TokenCredentialRequest)
+			if !ok {
+				plog.Debug(
+					"expected object to be TokenCredentialRequest",
+					"obj", plog.KObj(obj),
+					"actualType", fmt.Sprintf("%T", obj),
+				)
+				return
+			}
+
+			if tokenCredentialRequest.Spec.Authenticator.APIGroup == nil {
```

Comment:
OK I started implementing this and then ran into a question. The client should feel free to not set `tokenCredentialRequest.Spec.Authenticator.APIGroup`, right? So we should just `return nil` because there is nothing for us to mutate?

Reply:
Setting API group to `nil` is a bug that is only possible because we use `corev1.TypedLocalObjectReference` instead of a proper curated reference type.  We should error because making such a request is invalid.

Reply:
I _think_ I'm convinced by this statement in the other comment: `Note also that you could just purposefully use a kubeclient with no middleware (or a dynamic client) to get around any of these restrictions.`.

What I don't want to happen is for someone to be running Pinniped (server or client) in prod, and the `kubeclient` is dropping all `create TokenCredentialRequest` on the the floor because they don't have an optional field in the API set, and they can't do anything about it because the logic is burned into the executable. I guess we are saying "when adding `Middleware` to a `kubeclient`, you are agreeing to on all of the narrowed functionality".

So yeah, I guess I can return some errors here and for the other comment.

Reply:
61e3db81ec9bd1a7996cba7cbb3900655921e8fa